cmake/configs/base.cmake

```diff
@@ -212,11 +212,18 @@ config_bool(
     DEFAULT OFF
 )
 
+set(default_build_type "Debug")
+if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+  message(STATUS "Setting default build type to '${default_build_type}'.")
+  set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE
+      STRING "Type of build selected." FORCE)
+endif()
+
 set(default_optimize_level)
 if("${WT_OS}" STREQUAL "windows")
-    set(default_optimize_level "/O2")
+    set(default_optimize_level "/O0")
 else()
-    set(default_optimize_level "-O3")
+    set(default_optimize_level "-O0")
```

Comment:
Have you considered `-Og` for this? Not sure if MSVC has an equivalent.

Also, we should announce this change in a team channel. I (and I suspect a few other people) usually recycle a bunch of "known good" CMake flags whenever I'm building, so it's a chance to update people's assumptions about optimisation levels. (e.g. assuming optimised local builds could be bad when working on a server-related issue).

Reply:
Instead of /O0 for windows it might be /Od

https://docs.microsoft.com/en-us/cpp/build/reference/od-disable-debug?view=msvc-170

Reply:
I did look at `Og`, the clang documentation says:
```[-Og] Like [-O1]. In future versions, this option might disable different optimizations in order to improve debuggability.```

I'm inclined to go with the most extreme position here, but I could be convinced otherwise. I'd originally thought we were using the default of O1 here, but since it's O3 at the moment, O1 might be OK?

Ack re: Od - makes sense.

Reply:
I did a little experimenting, -O1 doesn't seem to optimise local variables away or inline stuff. Sorry to press the matter but I'd still prefer -O1 - completely turning off all optimisations seems like a bit too big of a hammer. (In my experience, -O1 will get you about 70% of the benefits of -O3 anyway.)

Reply:
I went and looked at the GCC docs, and they make a compelling case for -Og - I'll go ahead and make the change.

> -Og should be the optimization level of choice for the standard edit-compile-debug cycle, offering a reasonable level of optimization while maintaining fast compilation and a good debugging experience. It is a better choice than -O0 for producing debuggable code because some compiler passes that collect debug information are disabled at -O0.